sound/soc/intel/boards/sof_board_helpers.c

```diff
@@ -599,8 +600,8 @@ sof_intel_board_get_ctx(struct device *dev, unsigned long board_quirk)
 	if (!ctx)
 		return NULL;
 
-	ctx->codec_type = snd_soc_acpi_intel_detect_codec_type(dev);
-	ctx->amp_type = snd_soc_acpi_intel_detect_amp_type(dev);
+	ctx->codec_type = codec_type;
+	ctx->amp_type = amp_type;
```

Comment:
I am not sure I follow the benefits of this PR, it seems that we had ONE location where we detected the codec/amp type, and now we still have ONE location so this seems fairly equivalent but we've moved a lot of codec stuff into the parent PCI driver.

It's hard to see how this improves the code or helps support new capabilities. I am sure you have something in mind but looking only at the code it's hard to get the directions and improvements.